command/alloc_exec.go

```diff
@@ -0,0 +1,338 @@
+package command
+
+import (
+	"bytes"
+	"context"
+	"errors"
+	"fmt"
+	"io"
+	"os"
+	"os/signal"
+	"strings"
+	"syscall"
+
+	"github.com/docker/docker/pkg/term"
+	"github.com/hashicorp/nomad/api"
+	"github.com/hashicorp/nomad/api/contexts"
+	"github.com/posener/complete"
+)
+
+type AllocExecCommand struct {
+	Meta
+
+	Stdin  io.Reader
+	Stdout io.WriteCloser
+	Stderr io.WriteCloser
+}
+
+func (l *AllocExecCommand) Help() string {
+	helpText := `
+Usage: nomad alloc exec [options] <allocation> <command>
+
+  Run command inside the environment of the given allocation and task.
+
+General Options:
+
+  ` + generalOptionsUsage() + `
+
+Exec Specific Options:
+
+  -task <task-name>
+    Sets the task to exec command in
+
+  -job <job-id>
+    Use a random allocation from the specified job ID.
+
+  -i, --stdin=true
+    Pass stdin to the container, defaults to true
+
+  -t, --tty=true
+    Allocate a pseudo-tty, defaults to true if stdin is detected to be a tty session
+  `
+	return strings.TrimSpace(helpText)
+}
+
+func (l *AllocExecCommand) Synopsis() string {
+	return "Execute commands in task"
+}
+
+func (c *AllocExecCommand) AutocompleteFlags() complete.Flags {
+	return mergeAutocompleteFlags(c.Meta.AutocompleteFlags(FlagSetClient),
+		complete.Flags{
+			"--task":  complete.PredictAnything,
+			"-job":    complete.PredictAnything,
```

Comment:
We could predict the job here?

Reply:
Not quite - this matches the pattern of `nomad logs` where `--job` is boolean flag about whether to interpret the first argument is an alloc or a job id.  So `--job` itself doesn't take any arguments technically.

Reply:
Ah - In which case the documentation for `-job` needs to be updated to show that it's parsed as a boolean flag